webui/module/Job/view/job/job/index.phtml

```diff
@@ -167,19 +297,90 @@ $this->headTitle($title);
       return html.join('');
    }
 
-   function getJobs() {
+   function jobActionButtonsFormatter(value, row, index) {
+
+      let jobDetailsButton = '<a class="btn btn-default btn-xs" data-toggle="tooltip" data-placement="top" href="<?php echo $this->basePath(); ?>/job/details/'+row.jobid+'" title="<?php echo $this->translate("View Job Details"); ?>" id="btn-0"><span class="glyphicon glyphicon-search"></span></a>';
+      let jobRerunButton = '<a class="btn btn-default btn-xs" data-toggle="tooltip" data-placement="top" href="<?php echo $this->basePath() . '/job/index?action=rerun&jobid='; ?>'+row.jobid+'" title="<?php echo $this->translate('Rerun'); ?>" id="btn-1"><span class="glyphicon glyphicon-repeat"></span></a>';
+      let jobRestoreButton = '<a class="btn btn-default btn-xs" data-toggle="tooltip" data-placement="top" href="<?php echo $this->basePath(); ?>/restore/?mergefilesets=1&mergejobs=1&client='+row.client+'&jobid='+row.jobid+'" title="<?php echo $this->translate("Restore"); ?>" id="btn-1"><span class="glyphicon glyphicon-import"></span></a>';
+      let jobCancelButton = '<a class="btn btn-default btn-xs" data-toggle="tooltip" data-placement="top" href="<?php echo $this->basePath() . '/job/cancel/'; ?>'+row.jobid+'" title="<?php echo $this->translate("Cancel"); ?>" id="btn-1"><span class="glyphicon glyphicon-trash"></span></a>';
+
+      switch(row.jobstatus) {
+         case 'T':
+         case 'W':
+            switch(row.type) {
+               case 'B':
+                  return jobDetailsButton + '&nbsp;' + jobRerunButton + '&nbsp;' + jobRestoreButton;
+               case 'C':
+                  return jobDetailsButton + '&nbsp;' + jobRestoreButton;
+            }
```

Comment:
```suggestion
            }
            break
```

Reply:
I would suggest a break here, if no fallthrough is intended otherwise.

Reply:
good point